checker/match_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2018 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checker
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/envoyproxy/data-plane-api/api/auth"
+
+	"github.com/projectcalico/app-policy/proto"
+)
+
+// Successful parse should return name and namespace.
+func TestParseSpiffeIdOk(t *testing.T) {
+	RegisterTestingT(t)
+
+	id := "spiffe://foo.bar.com/ns/sandwich/sa/bacon"
+	name, namespace, err := parseSpiffeId(id)
+	Expect(name).To(Equal("bacon"))
+	Expect(namespace).To(Equal("sandwich"))
+	Expect(err).To(BeNil())
+}
+
+// Unsuccessful parse should return an error.
+func TestParseSpiffeIdFail(t *testing.T) {
+	RegisterTestingT(t)
+
+	id := "http://foo.bar.com/ns/sandwich/sa/bacon"
+	_, _, err := parseSpiffeId(id)
+	Expect(err).ToNot(BeNil())
+}
+
+// If no service account names are given, the clause matches any name.
+func TestMatchServiceAccountName(t *testing.T) {
+	RegisterTestingT(t)
+	testCases := []struct {
+		title  string
+		names  []string
+		name   string
+		result bool
+	}{
+		{"empty", []string{}, "reginald", true},
+		{"match", []string{"susan", "jim", "reginald"}, "reginald", true},
+		{"no match", []string{"susan", "jim", "reginald"}, "steven", false},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.title, func(t *testing.T) {
```

Comment:
one last thing, you need to move the `RegisterTestingT(t)` inside this `t.Run` sub-test otherwise when these tests fail, they will just panic and not give you the nice gomega assertion diff view.